pkg/ops/iso.go

```diff
@@ -415,11 +423,21 @@ func (b BuildISOAction) createEFI(rootdir string, isoDir string) error {
 	b.cfg.Logger.Infof("Detected Flavor: %s", flavor)
 	if strings.Contains(strings.ToLower(flavor), "ubuntu") {
 		b.cfg.Logger.Infof("Ubuntu based ISO detected, copying grub.cfg to /EFI/ubuntu/grub.cfg")
+		err = utils.MkdirAll(b.cfg.Fs, filepath.Join(temp, "EFI/ubuntu/"), constants.DirPerm)
+		if err != nil {
+			b.cfg.Logger.Errorf("Failed writing grub.cfg: %v", err)
+			return err
+		}
 		err = utils.MkdirAll(b.cfg.Fs, filepath.Join(isoDir, "EFI/ubuntu/"), constants.DirPerm)
 		if err != nil {
 			b.cfg.Logger.Errorf("Failed writing grub.cfg: %v", err)
 			return err
 		}
+		err = b.cfg.Fs.WriteFile(filepath.Join(temp, "EFI/ubuntu/", constants.GrubCfg), []byte(constants.GrubEfiCfg), constants.FilePerm)
```

Comment:
This is a nice trivia.

So when you build grub efi, you can set where it looks for the config rigth?

Most "normal" distros, are not stupid, so they set the grub.cfg search in the same dir as the grub efi file, so potentially you could have different grub versions with different grub configs.

/EFI/grub2/grub2.efi which will read the cfg file at /EFI/grub2/grub.cfg
/EFI/grub2-beta/grub2.efi which will read the cfg file at /EFI/grub2-beta/grub.cfg


BUT! Ubuntu is special, because why not?
So they hardcode where the grub.cfg file is and set a generic path in which to search for it, which is /EFI/ubuntu/grub.cfg

So we have to account for that and make sure that if the distro is ubuntu, we have to set the grub at that specific location.......yeah.

Reply:
🤦 ubuntu

thanks for clarifying